doc/reference/reference_lua/config/utils_schema.rst

```diff
@@ -0,0 +1,1163 @@
+        an error is raised.
+
+        The method performs the following checks:
+
+        -   field type checks: field values are checked against the schema node types
+        -   allowed values: if a node has the ``allowed_values`` annotations of schema nodes,
+            the corresponding data field is checked against the allowed values list
+        -   validation functions: if a validation function is defined for a node
+            (the ``validate`` annotation), it is executed to check that the provided value is valid.
+
+        :param any data: data
+
+        **Example:** see :ref:`config_utils_schema_annotation` and
+        :ref:`config_utils_schema_validating_configuration`
+
+        **See also:** :ref:`allowed_values <config-schema_node_annotation-allowed_values>`,
+        :ref:`validate <config-schema_node_annotation-validate>`
+
+    ..  _config-schema_object-methods:
+
+    ..  data:: methods
+
+        User-defined methods in the schema.
+
+        **See also:** :ref:`config_utils_schema_methods`
+
+    ..  _config-schema_object-name:
+
+    ..  data:: name
+
+        Schema name.
+
+    ..  _config-schema_object-schema:
+
+    ..  data:: schema
+
+        Schema nodes hierarchy.
+
+        **See also:** :ref:`config-utils-schema_node_object`
+
+
+..  _config-utils-schema_node_annotation:
+
+schema_node_annotation
+~~~~~~~~~~~~~~~~~~~~~~
+
+The following elements of tables passed as node constructor arguments are
+parsed by the modules as :ref:`built-in annotations <config_utils_schema_built_in_annotations>`:
+
+..  _config-schema_node_annotation-allowed_values:
+
+-   ``allowed_values``
+
+    A list of allowed values for a node.
+
+    **See also:** :ref:`schema_object:validate() <config-schema_object-validate>`
+
+
+..  _config-schema_node_annotation-apply_default_if:
+
+-   ``apply_default_if``
+
+    A boolean function that defines whether to apply the default value specified
+    using ``default``. If this function returns ``true`` on a provided configuration data,
+    the node receives the default value upon the :ref:`schema_object.apply_default() <config-schema_object-apply_default>`
+    method call.
+
+    The function takes two arguments:
+
+    -   ``data`` -- the configuration data
+    -   ``w`` -- *walkthrough node* with the following fields:
+
+        -   ``w.schema`` -- schema node
+        -   ``w.path`` -- the path to the schema node
+        -   ``w.error()`` -- a function for printing human-readable error messages
+
+    **See also:** :ref:`schema_object:apply_default() <config-schema_object-apply_default>`
+
+..  _config-schema_node_annotation-default:
+
+-   ``default``
+
+    A default value to use for a scalar node if it's not specified explicitly.
+
+    **Example:** see :ref:`config_utils_schema_transform_configuration`
+
+    **See also:** :ref:`schema_object:apply_default() <config-schema_object-apply_default>`
+
+..  _config-schema_node_annotation-type:
+
+-   ``type``
+
+    A value type.
```

Comment:
Maybe we should be a bit more precise here and say 'a schema node type' instead. A value with type `number` may be accepted by a node with type `number`, `integer`, `any` (and `string, number`, `number, string` as well). These are different things.

Reply:
changed as suggested